test/core/channel/channelz_registry_test.cc

```diff
@@ -43,56 +43,151 @@ namespace grpc_core {
 namespace channelz {
 namespace testing {
 
-TEST(ChannelzRegistryTest, UuidStartsAboveZeroTest) {
-  BaseNode* channelz_channel = nullptr;
-  intptr_t uuid = ChannelzRegistry::Register(channelz_channel);
+class ChannelzRegistryPeer {
+ public:
+  const InlinedVector<BaseNode*, 20>* entities() {
+    return &ChannelzRegistry::Default()->entities_;
+  }
+  int num_empty_slots() {
+    return ChannelzRegistry::Default()->num_empty_slots_;
+  }
+};
+
+class ChannelzRegistryTest : public ::testing::Test {
+ protected:
+  // ensure we always have a fresh registry for tests.
+  void SetUp() override { ChannelzRegistry::Init(); }
+
+  void TearDown() override { ChannelzRegistry::Shutdown(); }
+};
+
+TEST_F(ChannelzRegistryTest, UuidStartsAboveZeroTest) {
+  UniquePtr<BaseNode> channelz_channel(
```

Comment:
Could use `MakeUnique<>` here.

Same thing in all of these tests.

Reply:
Done